velox/connectors/hive/storage_adapters/hdfs/HdfsWriteFile.cpp

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/connectors/hive/storage_adapters/hdfs/HdfsWriteFile.h"
+#include <hdfs/OutputStream.h>
+#include <hdfs/hdfs.h>
+
+namespace facebook::velox {
+
+HdfsWriteFile::HdfsWriteFile(hdfsFS hdfs, std::string_view path) {
+  // Pass 0 to the parameters bufferSize/replication/blockSize, means to use the
+  // default configured value. This is commented in the hdfs.h/hdfsOpenFile
+  // function.
+  new (this) HdfsWriteFile(hdfs, path, 0, 0, 0);
+}
+
+HdfsWriteFile::HdfsWriteFile(
+    hdfsFS hdfs,
+    std::string_view path,
+    int bufferSize,
+    short replication,
+    int blockSize)
+    : hdfsClient_(hdfs), filePath_(path) {
+  hdfsFile_ = hdfsOpenFile(
+      hdfsClient_,
+      filePath_.c_str(),
+      Hdfs::CreateFlag::Create | Hdfs::CreateFlag::Append,
+      bufferSize,
+      replication,
+      blockSize);
+  VELOX_CHECK_NOT_NULL(
+      hdfsFile_,
+      "Failed to open hdfs file: {}, with error: {}",
+      filePath_,
+      std::string(hdfsGetLastError()));
+}
+
+void HdfsWriteFile::close() {
+  int flag = hdfsCloseFile(hdfsClient_, hdfsFile_);
+  VELOX_CHECK_EQ(
+      flag,
+      0,
+      "Failed to close hdfs file, with error:{}",
+      std::string(hdfsGetLastError()));
+}
+
+void HdfsWriteFile::flush() {
+  int success = hdfsFlush(hdfsClient_, hdfsFile_);
+  VELOX_CHECK_EQ(
+      success, 0, "Hdfs flush error: {}", std::string(hdfsGetLastError()));
+}
+
+void HdfsWriteFile::append(std::string_view data) {
+  write(data, data.size());
+}
+
+void HdfsWriteFile::write(std::string_view data, int64_t dataSize) {
+  int64_t totalBytesWrite = 0;
+  if (data.size() != 0) {
```

Comment:
If the file is not empty, then calling write with dataSize being zero won't make the file empty. Is this expected?

Reply:
If the HdfsWriteFile not closed and calling the function by append with dataSize 0, then nothing would be append,  the file will not be empty, I think this is normal. And If the HdfsWriteFile is closed and reopened, it will be empty when reopen it, at this moment call this function with dataSize